### DISCUSSION TITLE
Missing / incomplete tiles where rasters' valid data borders each other

### DISCUSSION THREAD

**Discussion by zstatmanweil** [stac-utils/titiler-pgstac#184] [Q&A] [answered]
I am working with rasters that correspond to watersheds and so have valid data within these boundaries and nodata without. When Rasters are float32, 5m resolution. I view them with `titiler-pgstac` at zoom 11, I have no issues:

![image](https://github.com/user-attachments/assets/3a19c147-e5dc-463d-9d9f-19c56a3ab9ff)

When I zoom in, I start seeing oddities like here you can see the tile is only filled in by one of the watersheds values:

![image](https://github.com/user-attachments/assets/61371a8a-2ff1-4c7e-a9c6-bd1314aea76e)

I zoom in more and some tiles are completely missing data:
![image](https://github.com/user-attachments/assets/cf6c2f76-9f77-4be6-aa29-ecc13facdf96)

I tried to play around with `pixel_selection`, `resampling` and `scale` but the same issue occurred in the same places. 

I am a little unclear on which DEBUG env variable name I should use so I tried all three of these but none seemed to actually show any logs:
      - DEBUG=TRUE
      - TITILER_DEBUG=TRUE
      - TITILER_PGSTAC_API_DEBUG=TRUE

Any pointers on troubleshooting would be much appreciated!

**Comment by drnextgis** [comment]
If you use titiler-pgstac which comes with eoAPI then to enable DEBUG you should use [EOAPI_RASTER_DEBUG](https://github.com/developmentseed/eoapi-devseed/blob/86545b0817f369d4d8b1587085809332a020d79d/runtimes/eoapi/raster/eoapi/raster/config.py#L18).

**Comment by zstatmanweil** [comment]
Thanks @drnextgis, I am not running via eoAPI though

**Comment by zstatmanweil** [comment, marked as answer]
When I set both `exitwhenfull` and `skipcovered` to False, I no longer have this issue. I think I was assuming it would factor in the nodata value but since those parameters are used on the pgstac side, it makes sense it is just looking at the geometry, not assessing the raster or its nodata value. 

No issues here!

**Reply by vincentsarago** [reply]
Glad you found a solution! 

I guess your STAC Items geometry cover a larger area than the dataset valid data! We had this issue with Landsat data in the past, when creating the Item we would get the geometry from the bbox of the data instead matching the valid data (rotated) (like https://github.com/developmentseed/rio-stac/issues/45)

**Reply by zstatmanweil** [reply]
Yes, exactly. Our valid data matches watershed boundaries but our STAC item geometries are just the entire raster's footprint. So a somewhat niche problem

**Reply by vincentsarago** [reply]
maybe we should add settings to configure https://github.com/stac-utils/titiler-pgstac/blob/d353b9df7b8b2ff164258d94c014386a34d3ec89/titiler/pgstac/mosaic.py#L276-L280

**Reply by zstatmanweil** [reply]
Do you mean so they don't have to passed as query params every time? In my use case that would definitely be useful

**Reply by vincentsarago** [reply]
yes

**Reply by zstatmanweil** [reply]
Thanks for opening an issue. I may have time to work on it.
